### ISSUE TITLE
Planning for release of v5.3.2

### ISSUE THREAD

**Issue by Jermolene** [Jermolene/TiddlyWiki5#7825]
The planned schedule for the release of v5.3.2 is as follows:

* Monday ~~13th November 2023~~ 27th November 2023- PR merging freeze, apart from docs and bug fixes. This is also when I will kick off the banner artwork competition
* Monday ~~27th November 2023~~ 11th December - Release of v5.3.2

I am hoping to merge the following PRs before the freeze:

* #7855
* #7849
* #7843
* #7649
* #7769
* #7694
* #7551 
* #7776 
* #7634

Deferred – see #7856 for details

* #7817
* #7830
* #7822 
* #7615 
* #7548 
* #7729
* #7734

Please let me know here if there are other PRs for consideration.

**Comment by CrossEye** [comment]
I don't know if #7729 is ready from your perspective.  It feels complete to me.  If so, I hope it's up for consideration.

**Comment by rmunn** [comment]
I want to get #7694 in. It's been ready for a while, just need to work out the best way to add a `$list-join` widget to parallel the `$list-template` widget. Which means getting #7804 fixed. (I think fixing #7804 before release is really important, otherwise the current surprising behavior will become locked in by backwards-compatibility policies, and I don't think that's what we want.)

**Comment by pmario** [comment]
Please check: **@pmario's Open Pull Requests** #7657

**Comment by pmario** [comment]
- #7822

**Comment by pmario** [comment]
- #7776

**Comment by pmario** [comment]
- #7830

**Comment by rmunn** [comment]
- #7827 is ready to merge
- Once that one is merged, #7694 will be ready to go, possibly after I solve an easy merge conflict

**Comment by rmunn** [comment]
Doing extensive tests on #7694, and found a [weird corner-case bug](https://github.com/Jermolene/TiddlyWiki5/pull/7694#issuecomment-1798059144) as well as a [simpler reproduction for it](https://github.com/Jermolene/TiddlyWiki5/pull/7694#issuecomment-1798076403) (though that might be a different bug). No more merge conflicts on #7694, but I need to nail down those bugs before I feel confident in merging it. If #7694 misses the merge window for 5.3.2, oh well, it can go into 5.3.3. I really want to get it into 5.3.2, though, so I'll keep you updated on my progress nailing down those last couple of bugs.

**Comment by rmunn** [comment]
- #7694 is now ready to merge. All bugs found so far are fixed, and I've been unable to turn up any more bugs in quite extensive testing.

**Comment by Jermolene** [comment]
Thanks @CrossEye @rmunn @pmario I've updated the OP with the tickets you've mentioned.

**Comment by rmunn** [comment]
- https://github.com/Jermolene/TiddlyWiki5/pull/7835 needs to be merged before #7694. It's a fix for the parsing bug I discovered that was causing list joins to come out as "1and 2and 3" instead of "1 and 2 and 3" (i.e., the first space character was incorrectly being skipped).

**Comment by Jermolene** [comment]
I've updated the roadmap in the OP to push the release out by 2 weeks. The reason is that I have been working on #7837 and would like to try to get a fix into the next release.

**Comment by rmunn** [comment]
Then maybe I'll work on PRs for #7839 and #7840 (negative indexes, and adding `jsoninsert`, `jsonremove`, `jsonpush`, and `jsonpop` operators). They should be straightforward; with a couple extra weeks to work with I should be able to get them done before the new PR freeze date.

**Comment by Jermolene** [comment]
> I've updated the roadmap in the OP to push the release out by 2 weeks. The reason is that I have been working on #7837 and would like to try to get a fix into the next release.

The draft PR is at #7843

> Then maybe I'll work on PRs for #7839 and #7840 (negative indexes, and adding `jsoninsert`, `jsonremove`, `jsonpush`, and `jsonpop` operators). They should be straightforward; with a couple extra weeks to work with I should be able to get them done before the new PR freeze date.

Great, thanks @rmunn

**Comment by rmunn** [comment]
https://github.com/Jermolene/TiddlyWiki5/pull/7849 is a PR for #7839 (allowing negative indexes in JSON operators). I'll defer implementing #7840 until the 5.3.3 release cycle, so that there's time to have a conversation about the best set of operators to implement.

**Comment by Jermolene** [comment]
I've created the usual ticket to track PRs that have been deferred from this release: #7856

**Comment by pmario** [comment]
@Jermolene ... It seems the current timeline "stands". right? -- I would like to create a post at Talk, to let the users know, that they should start extensive testing of the prerelease version. I'll pin the post till 8th. Dec 23

**Comment by Jermolene** [comment]
> @Jermolene ... It seems the current timeline "stands". right? -- I would like to create a post at Talk, to let the users know, that they should start extensive testing of the prerelease version. I'll pin the post till 8th. Dec 23

Yes, the freeze is now in place, albeit @saqimtiaz and I plan to continue work on #7734 for a few days because it doesn't affect the core itself.

I plan to make an announcement on Talk shortly.

**Comment by Jermolene** [comment]
> Yes, the freeze is now in place, albeit @saqimtiaz and I plan to continue work on #7734 for a few days because it doesn't affect the core itself.

We've decided to defer the improvements to the Tour plugin until the next release. There are no other PRs planned for merging but  we will continue to accept translation updates (and docs updates, as always).

**Comment by Jermolene** [comment]
Closing this now that v5.3.2 has been released
